optimizely/optimizely.py

```diff
@@ -1130,7 +1194,11 @@ def _decide(self, user_context, key, decide_options=None):
                                   user_context=user_context, reasons=reasons if should_include_reasons else []
                                   )
 
-    def _decide_all(self, user_context, decide_options=None):
```

Comment:
is OptimizelyUserContext an optional parameter?

Reply:
Currently `_decide_all` is called from `decide_all` using `OptimizelyUserContext._clone` which has a return type of `Optional[OptimizelyUserContext]`. The `None` case would be handled by line 1212, but if you think we should add an additional check to the `decide_all` method that handles `_clone` returning `None`, that would allow us to make the `OptimizelyUserContext` not optional.

The same thing goes for `_decide` and `_decide_for_keys`.